File: MCTabBarDemo/ViewController.m
//
//  ViewController.m
//  MCTabBarDemo
//
//  Created by chh on 2017/12/4.
//  Copyright © 2017年 chh. All rights reserved.
//

#import "ViewController.h"
#import "PushViewController.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIColor *randomColor = [UIColor colorWithRed:(arc4random()%255)*1.0f/255.0
                                                  green:(arc4random()%255)*1.0f/255.0
                                                   blue:(arc4random()%255)*1.0f/255.0 alpha:0.6];
    self.view.backgroundColor = randomColor;
    
    UIButton *btn = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 150, 80)];
    btn.center = self.view.center;
    [btn setTitle:@"按钮" forState:UIControlStateNormal];
    [btn setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [btn setBackgroundColor:[UIColor orangeColor]];
    btn.layer.cornerRadius = 4.0;
    [btn addTarget:self action:@selector(btnAction) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
}

- (void)btnAction{
    
    [self.navigationController pushViewController:[[PushViewController alloc] init] animated:YES];
   
}
@end

File: MCTabbarDemoSwift/ViewController.swift
//
//  ViewController.swift
//  MCTabbarDemoSwift
//
//  Created by caohouhong on 2018/12/4.
//  Copyright © 2018年 caohouhong. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let randomColor = UIColor(red: CGFloat((arc4random()%255))/255.0, green: CGFloat((arc4random()%255))/255.0, blue: CGFloat((arc4random()%255))/255.0, alpha: 1)
        self.view.backgroundColor = randomColor
        
        let button = UIButton(frame: CGRect(x: 0, y: 0, width: 150, height: 80))
        button.center = self.view.center
        button.layer.cornerRadius = 5.0
        button.setTitle("按钮", for: .normal)
        button.setTitleColor(UIColor.white, for: .normal)
        button.backgroundColor = UIColor.orange
        button.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
        self.view.addSubview(button)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    @objc func buttonAction(){
        self.navigationController?.pushViewController(PushViewController(), animated: true)
    }
}

